### PR TITLE
enhance(template): add support for global environment

### LINKED ISSUE #410: Enable templates to return global environment
## Description

[Templates](https://go-vela.github.io/docs/templates/) should support the ability to export the global [environment](https://go-vela.github.io/docs/reference/yaml/environment/) block.

## Value

Feature parity with the base `.vela.yml`. 

## Definition of Done

Any template is able to return the global environment block.

## Effort  (Optional)

4 hours.

## Impacted Personas (Optional)

All templates.

### PR DESCRIPTION
**Pull request by JordanSussman**
closes https://github.com/go-vela/community/issues/410

### CODE REVIEW

File: compiler/native/testdata/template-starlark.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "size": 5362,
   "name": "template.star",
   "path": "template.star",
-  "content": "ZGVmIG1haW4oY3R4KToKICByZXR1cm4gewogICAgJ3ZlcnNpb24nOiAnMScsCiAgICAnc3RlcHMnOiBbCiAgICAgIHsKICAgICAgICAnbmFtZSc6ICdidWlsZCcsCiAgICAgICAgJ2ltYWdlJzogJ2dvbGFuZzpsYXRlc3QnLAogICAgICAgICdjb21tYW5kcyc6IFsKICAgICAgICAgICdnbyBidWlsZCcsCiAgICAgICAgICAnZ28gdGVzdCcsCiAgICAgICAgXQogICAgICB9LAogICAgXSwKfQ==",
+  "content": "ZGVmIG1haW4oY3R4KToKICByZXR1cm4gewogICAgJ3ZlcnNpb24nOiAnMScsCiAgICAnZW52aXJvbm1lbnQnOiB7CiAgICAgICdzdGFyJzogJ3Rlc3QzJywKICAgICAgJ2Jhcic6ICd0ZXN0NCcsCiAgICB9LAogICAgJ3N0ZXBzJzogWwogICAgICB7CiAgICAgICAgJ25hbWUnOiAnYnVpbGQnLAogICAgICAgICdpbWFnZSc6ICdnb2xhbmc6bGF0ZXN0JywKICAgICAgICAnY29tbWFuZHMnOiBbCiAgICAgICAgICAnZ28gYnVpbGQnLAogICAgICAgICAgJ2dvIHRlc3QnLAogICAgICAgIF0KICAgICAgfSwKICAgIF0sCn0K\n",
```
Comment (JordanSussman) [inline compiler/native/testdata/template-starlark.json:7]: You can view the output of this with `echo '<content>' | base64 -D`.

File: compiler/native/testdata/template.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
...
...ogInRlc3Q0IgoKbWV0YWRhdGE6CiAgdGVtcGxhdGU6IHRydWUKCnN0ZXBzOgogIC0gbmFtZTogaW5zdGFsbAogICAgY29tbWFuZHM6CiAgICAgIC0gLi9ncmFkbGV3IGRvd25sb2FkRGVwZW5kZW5jaWVzCiAgICBlbnZpcm9ubWVudDoge3sgLmVudmlyb25tZW50IH19CiAgICBpbWFnZToge3sgLmltYWdlIH19CiAgICB7eyAucHVsbF9wb2xpY3kgfX0KCiAgLSBuYW1lOiB0ZXN0CiAgICBjb21tYW5kczoKICAgICAgLSAuL2dyYWRsZXcgY2hlY2sKICAgIGVudmlyb25tZW50OiB7eyAuZW52aXJvbm1lbnQgfX0KICAgIGltYWdlOiB7eyAuaW1hZ2UgfX0KICAgIHt7IC5wdWxsX3BvbGljeSB9fQoKICAtIG5hbWU6IGJ1aWxkCiAgICBjb21tYW5kczoKICAgICAgLSAuL2dyYWRsZXcgYnVpbGQKICAgIGVudmlyb25tZW50OiB7eyAuZW52aXJvbm1lbnQgfX0KICAgIGltYWdlOiB7eyAuaW1hZ2UgfX0KICAgIHt7IC5wdWxsX3BvbGljeSB9fQoKc2VjcmV0czoKICAtIG5hbWU6IGRvY2tlcl91c2VybmFtZQogICAga2V5OiBvcmcvcmVwby9mb28vYmFyCiAgICBlbmdpbmU6IG5hdGl2ZQogICAgdHlwZTogcmVwbwoKICAtIG5hbWU6IGZvb19wYXNzd29yZAogICAga2V5OiBvcmcvcmVwby9mb28vcGFzc3dvcmQKICAgIGVuZ2luZTogdmF1bHQKICAgIHR5cGU6IHJlcG8KCnt7IGlmIC5zZWNyZXQgfX0KCiAgLSBuYW1lOiBiYXJfcGFzc3dvcmQKICAgIGtleTogb3JnL3JlcG8vYmFyL3Bhc3N3b3JkCiAgICBlbmdpbmU6IHZhdWx0CiAgICB0eXBlOiByZXBvCgp7eyBlbmQgfX0KCnNlcnZpY2VzOgogICAtIG5hbWU6IHBvc3RncmVzCiAgICAgaW1hZ2U6IHBvc3RncmVzOjEyCgoge3sgaWYgLnNlcnZpY2UgfX0KCiAgIC0gbmFtZTogcmVkaXMKICAgICBrZXk6IHJlZGlzOjYKCiB7eyBlbmQgfX0K",
```
Comment (JordanSussman) [inline compiler/native/testdata/template.json:7]: You can view the output of this with `echo '<content>' | base64 -D`.

**Review by jbrockopp — APPROVED** [review]
LGTM

**Review by kneal — APPROVED** [review]
LGTM 🐬

**Pull request merged**